virtio-spec/src/virtq/mod.rs

```diff
@@ -0,0 +1,255 @@
+//! Virtqueue definitions
+
+#[cfg(feature = "alloc")]
+mod alloc;
+
+use core::alloc::Layout;
+use core::ptr::{addr_of_mut, NonNull};
+use core::{mem, ptr};
+
+use endian_num::{le16, le32, le64};
+
+/// Split Virtqueue Descriptor
+#[doc(alias = "virtq_desc")]
+#[derive(Clone, Copy, Debug)]
+#[repr(C)]
+pub struct Desc {
+    /// Address (guest-physical).
+    pub addr: le64,
+
+    /// Length.
+    pub len: le32,
+
+    /// The flags as indicated in [`DescF`].
+    pub flags: DescF,
+
+    /// Next field if flags & NEXT
+    pub next: le16,
+}
+
+endian_bitflags! {
+    /// Virtqueue descriptor flags
+    #[doc(alias = "VIRTQ_DESC_F")]
+    pub struct DescF: le16 {
+        /// This marks a buffer as continuing via the next field.
+        #[doc(alias = "VIRTQ_DESC_F_NEXT")]
+        const NEXT = 1;
+
+        /// This marks a buffer as device write-only (otherwise device read-only).
+        #[doc(alias = "VIRTQ_DESC_F_WRITE")]
+        const WRITE = 2;
+
+        /// This means the buffer contains a list of buffer descriptors.
+        #[doc(alias = "VIRTQ_DESC_F_INDIRECT")]
+        const INDIRECT = 4;
+
+        #[doc(alias = "VIRTQ_DESC_F_AVAIL")]
+        const AVAIL = 1 << 7;
+
+        #[doc(alias = "VIRTQ_DESC_F_USED")]
+        const USED = 1 << 15;
+    }
+}
+
+/// The Virtqueue Available Ring
+#[doc(alias = "virtq_avail")]
+#[derive(Debug)]
+#[repr(C)]
+pub struct Avail {
```

Comment:
While it might be nicer, this crate will always use the naming from the spec: `virtq_avail` -> `virtq::Avail`.

It can, of course, always be imported with a different name, but for this PR, I'd like to stick to the spec in the kernel as well.